app/javascript/components/applications/application_form/PrimaryApplicantSection.js

```diff
@@ -15,15 +16,46 @@ let mailingAddressFieldMap = {
   zip: 'mailingZip',
 }
 
+const validates = (fun, message) => (value) => {
+  return fun(value) ? null : message
+}
+
+const isOldEnough = (dateOfBirth) => {
+  const years = moment().diff(dateOfBirth,'years')
+  return years >= 18
+}
+
+const FormError = ({formApi, field }) => {
+  if ((formApi.touched[field] || formApi.submitted) && formApi.errors[field])
+    return <span className="small error">{formApi.errors[field]}</span>
+  else
+    return null
+}
+
+const errorClassName = (formApi, field) => {
+  if (formApi.touched[field] || formApi.submitted)
+    return { error: !!formApi.errors[field] }
+  else
+    return null
+}
```

Comment:
are these functions general enough that they could be pulled into a util file or something?

Reply:
yes, actually I'm working on a different story with validation and pulling this out

Reply:
on master or on caritas?

Reply:
caritas